execext/exec.go

```diff
@@ -2,22 +2,49 @@ package execext
 
 import (
 	"context"
-	"os/exec"
+	"errors"
+	"io"
+	"strings"
+
+	"github.com/mvdan/sh/interp"
+	"github.com/mvdan/sh/syntax"
 )
 
+type RunCommandOptions struct {
+	Command string
+	Dir     string
+	Env     []string
+	Stdin   io.Reader
+	Stdout  io.Writer
+	Stderr  io.Writer
+}
+
 var (
-	// ShPath is path to "sh" command
-	ShPath string
-	// ShExists is true if "sh" command is available on the system
-	ShExists bool
+	// ErrNilOptions is returned when a nil options is given
+	ErrNilOptions = errors.New("execext: nil options given")
 )
 
-func init() {
-	var err error
-	ShPath, err = exec.LookPath("sh")
-	ShExists = err == nil
-}
+// RunCommand runs a shell command
+func RunCommand(ctx context.Context, opts *RunCommandOptions) error {
+	if opts == nil {
+		return ErrNilOptions
+	}
+
+	p, err := syntax.Parse(strings.NewReader(opts.Command), "", syntax.ParseMode(0))
```

Comment:
It also does other things, like tell the parser whether or not to keep comments: https://godoc.org/github.com/mvdan/sh/syntax#ParseMode

In the future (2.0) I think I'll change this, because mixing in such different things in a bitfield is a bit confusing.